staging/src/k8s.io/component-base/metrics/legacyregistry/registry.go

```diff
@@ -0,0 +1,89 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package legacyregistry
+
+import (
+	apimachineryversion "k8s.io/apimachinery/pkg/version"
+	"k8s.io/component-base/metrics"
+	"sync"
+)
+
+var globalRegistryFactory = metricsRegistryFactory{
+	registerQueue:     make([]metrics.KubeCollector, 0),
+	mustRegisterQueue: make([]metrics.KubeCollector, 0),
+}
+
+type metricsRegistryFactory struct {
+	globalRegistry    metrics.KubeRegistry
+	kubeVersion       *apimachineryversion.Info
+	registrationLock  sync.Mutex
+	registerQueue     []metrics.KubeCollector
+	mustRegisterQueue []metrics.KubeCollector
+}
+
+// SetRegistryFactoryVersion sets the kubernetes version information for all
+// subsequent metrics registry initializations. Only the first call has an effect.
```

Comment:
I wouldn't mind to see a panic on the second call, or at least a klog.Warning.

Reply:
done.